src/main/java/ua/edu/internship/interview/service/business/InterviewService.java

```diff
@@ -0,0 +1,152 @@
+        log.info("Updated feedback for interview with id: {}", updatedInterview.getId());
+        return interviewMapper.toDto(updatedInterview);
+    }
+
+    public InterviewQuestionDto createInterviewQuestion(String interviewId, InterviewQuestionCreateDto dto) {
+        log.info("Attempting to create interview question for interview with id: {}", interviewId);
+        InterviewQuestionDocument questionDocument = questionMapper.toDocument(dto);
+        UserQuestionDocument userQuestionDocument = getUserQuestionByIdOrElseThrow(dto.getUserQuestionId());
+        questionDocument.setQuestion(userQuestionDocument);
+
+        InterviewDocument interviewDocument = getInterviewByIdOrElseThrow(interviewId);
+        interviewDocument.addQuestion(questionDocument);
+        interviewRepository.save(interviewDocument);
+        log.info("Created interview question for interview with id: {}", interviewDocument.getId());
+        return questionMapper.toDto(questionDocument);
+    }
+
+    public InterviewQuestionDto updateInterviewQuestion(String interviewId, String questionId, InterviewQuestionUpdateDto dto) {
+        log.info("Attempting to update interview question for interview with id: {}", interviewId);
+        InterviewDocument interviewDocument = getInterviewByIdOrElseThrow(interviewId);
+        InterviewQuestionDocument questionDocument = getInterviewQuestionOrElseThrow(questionId, interviewDocument);
+        questionMapper.updateDocument(dto, questionDocument);
+        interviewRepository.save(interviewDocument);
+        log.info("Updated interview question for interview with id: {}", interviewDocument.getId());
+        return questionMapper.toDto(questionDocument);
+    }
+
+    public void deleteInterviewQuestionById(String interviewId, String questionId) {
+        log.info("Attempting to delete interview question for interview with id: {}", interviewId);
+        InterviewDocument interviewDocument = getInterviewByIdOrElseThrow(interviewId);
+        interviewDocument.getQuestions().removeIf(q -> q.getId().equals(questionId));
+        interviewRepository.save(interviewDocument);
+        log.info("Deleted interview question for interview with id: {}", interviewId);
+    }
+
+    private void validateInterviewNotCausingConflicts(InterviewDocument interview) {
+        LocalDateTime plannedTime = interview.getPlannedTime();
+        LocalDateTime from = plannedTime.minusHours(1);
+        LocalDateTime to = plannedTime.plusHours(1);
+        List<InterviewDocument> conflictingInterviews = interviewRepository
+                .findInterviewsInTimeWindow(interview.getInterviewerId(), interview.getCandidateId(), from, to);
+        if (!conflictingInterviews.isEmpty()) {
+            String exceptionMessage = String.format(
+                    "Cannot create interview for interviewer '%s' and candidate '%s' at %s. It overlaps existing interviews",
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 120 characters (found 125).
<!-- __reviewdog__:ChAyYjQwZWI2OWI1Yjc3Y2EwEgpjaGVja3N0eWxl -->